tests/test_projects/registers/mod.rs

```diff
@@ -23,63 +26,55 @@ async fn can_get_overflow() {
 
 #[tokio::test]
 async fn can_get_program_counter() {
-    abigen!(
-        TestFuelCoinContract,
-        "test_projects/registers/out/debug/registers-abi.json",
-    );
     let salt = Salt::from([0u8; 32]);
-    let compiled = Contract::compile_sway_contract("test_projects/registers", salt).unwrap();
+    let compiled =
+        Contract::load_sway_contract("test_projects/registers/out/debug/registers.bin", salt)
+            .unwrap();
     let client = Provider::launch(Config::local_node()).await.unwrap();
     let id = Contract::deploy(&compiled, &client).await.unwrap();
     let instance = TestFuelCoinContract::new(id.to_string(), client);
 
     let result = instance.get_program_counter().call().await.unwrap();
 
-    assert_eq!(result.value, 1760);
+    assert_eq!(result.value, 1728);
 }
 
 #[tokio::test]
 async fn can_get_stack_start_ptr() {
-    abigen!(
-        TestFuelCoinContract,
-        "test_projects/registers/out/debug/registers-abi.json",
-    );
     let salt = Salt::from([0u8; 32]);
-    let compiled = Contract::compile_sway_contract("test_projects/registers", salt).unwrap();
+    let compiled =
+        Contract::load_sway_contract("test_projects/registers/out/debug/registers.bin", salt)
+            .unwrap();
     let client = Provider::launch(Config::local_node()).await.unwrap();
     let id = Contract::deploy(&compiled, &client).await.unwrap();
     let instance = TestFuelCoinContract::new(id.to_string(), client);
 
     let result = instance.get_stack_start_ptr().call().await.unwrap();
 
-    assert_eq!(result.value, 2176);
+    assert_eq!(result.value, 1952);
 }
 
 #[tokio::test]
 async fn can_get_stack_ptr() {
-    abigen!(
-        TestFuelCoinContract,
-        "test_projects/registers/out/debug/registers-abi.json",
-    );
     let salt = Salt::from([0u8; 32]);
-    let compiled = Contract::compile_sway_contract("test_projects/registers", salt).unwrap();
+    let compiled =
+        Contract::load_sway_contract("test_projects/registers/out/debug/registers.bin", salt)
+            .unwrap();
     let client = Provider::launch(Config::local_node()).await.unwrap();
     let id = Contract::deploy(&compiled, &client).await.unwrap();
     let instance = TestFuelCoinContract::new(id.to_string(), client);
 
     let result = instance.get_stack_ptr().call().await.unwrap();
 
-    assert_eq!(result.value, 2176);
+    assert_eq!(result.value, 1952);
```

Comment:
Why did you have to change this number? If it's some hard-coded thing that may have to change as the compiler changes, seems hard to maintain?

Reply:
yeah, not sure the best approach to testing the return value from these registers. There are a bunch of hard-coded values that could change down the road.
For the values relating to gas, I may be able to subtract gas used from total gas for example, but that approach won't work for the stack pointers and other registers.
I could simply assert they are > 0 which should be easier to maintain...

Reply:
They should be > 0 and < the max memory address. Beyond that, nothing is really guaranteed without hard-coding.

Reply:
@adlerjohn what is the max memory address atm?

Reply:
https://github.com/FuelLabs/fuel-vm/blob/adc5b93958165329a1566d8173819cb92e3666d9/src/consts.rs#L76

Reply:
@adlerjohn I've added a less brittle check for some of these tests to avoid the hardcoded values.